src/main/java/org/tptacs/domain/entities/Order.java

```diff
@@ -1,11 +1,14 @@
 package org.tptacs.domain.entities;
 
 import lombok.Getter;
+import lombok.Setter;
+
 import org.tptacs.domain.enums.OrderStatus;
 import java.time.LocalDateTime;
 import java.util.List;
 
 @Getter
+@Setter
```

Comment:
Por qué rompemos el encapsulamiento de una entidad de dominio poniendo el @Setter a nivel de clase? Esto permitiría por ejemplo setear el id o userId, que son campos que no tiene sentido que puedan ser modificables.

Crearía un método close() que haga esto:

`public void close() {
this.status = OrderStatus.CLOSED
this.lastUpdate = LocaldDateTime.now()
}`

Reply:
no veia mal que tenga setter, si queres modificarle algo al objeto tenes que tener sus seter, pero bueno lo mudaremos al metodo close